include/oneapi/dpl/internal/histogram_extension_defs.h

```diff
@@ -0,0 +1,48 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_HISTOGRAM_EXTENSION_DEFS_H
+#define _ONEDPL_HISTOGRAM_EXTENSION_DEFS_H
+
+#include "../pstl/onedpl_config.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+
+// Only supported for dpcpp backend
+#ifdef _ONEDPL_BACKEND_SYCL
```

Comment:
Since at the moment we are only supporting the SYCL backend use case, would it be beneficial to have some sort of `static_assert` or graceful error if the user attempts to compile for a non-SYCL backend to prevent confusion?

Reply:
I looked at doing this (a few commits back in the history).  However, for some compilers it seems the function is instantiated and hits the `static_assert` even without an actual call of this function.  We could instead do it with a runtime assert, perhaps that is the way to go.